sdk/src/test/java/com/horizen/account/transaction/EthereumTransactionTest.java

```diff
@@ -351,5 +352,38 @@ public void ethereumSignedEIP1559TransactionTest() {
         assertArrayEquals(toEthereumTransaction.getTo().address(), toProposition.address());
     }
 
+    private RawTransaction[] generateTransactions(int count) {
+        var to = "0x0000000000000000000000000000000000001337";
+        var data = "01020304050607080a";
+        var gas = BigInteger.valueOf(21000);
+        var gasPrice = BigInteger.valueOf(2000000000);
+        var txs = new RawTransaction[count];
+        for (int i = 0; i < txs.length; i++) {
+            var nonce = BigInteger.valueOf(i);
+            var value = BigInteger.valueOf(i).multiply(BigInteger.TEN.pow(18));
+//            if (i % 3 == 0) {
+            txs[i] = RawTransaction.createTransaction(nonce, gasPrice, gas, to, value, data);
+//            } else {
+//                txs[i] = RawTransaction.createTransaction(0, nonce, gas, to, value, data, gasPrice, gasPrice);
+//            }
+        }
+        return txs;
+    }
 
+    @Test
+    public void ethereumTransactionsRootHashTest() {
+        final var testCounts = new int[]{0, 1, 2, 3, 4, 10, 51, 1000, 126, 127, 128, 129, 130, 765};
+        for (var count : testCounts) {
+            var txs = generateTransactions(count);
+            var rlpTxs = Arrays
+                    .stream(txs)
+                    .map(tx -> TransactionEncoder.encode(
+                            tx,
+                            // make sure we also encode a signature, even if it is empty to make the results comparable to GETH
+                            new Sign.SignatureData(new byte[0], new byte[0], new byte[0])
+                    ))
+                    .toArray(byte[][]::new);
+            System.out.format("transactions root hash (%4d) %s%n", count, Numeric.toHexString(TrieHasher.Root(rlpTxs)));
```

Comment:
Do we have some regression data in GETH tests to compare with?

Reply:
yes, added